src/context/WalletProvider/WalletProvider.tsx

```diff
@@ -83,15 +92,28 @@ const reducer = (state: InitialState, action: ActionTypes) => {
     case WalletActions.SET_ADAPTERS:
       return { ...state, adapters: action.payload }
     case WalletActions.SET_WALLET:
-      return {
+      const stateData = {
         ...state,
         wallet: action.payload.wallet,
         walletInfo: {
           name: action?.payload?.name,
           icon: action?.payload?.icon,
-          deviceId: action?.payload?.deviceId
+          deviceId: action?.payload?.deviceId,
+          meta: { label: '', address: '' }
         }
       }
+
+      if (action.payload.meta && action.payload.meta.label)
+        stateData.walletInfo.meta.label = action.payload.meta.label
```

Comment:
Nitpick for me but it might or might not be for other reviewers: don't mutate objects, it can probably be done in place
`meta: { label: action.payload.meta?.label ?? '', address: '' }` (probably cleaner and more concise)

Reply:
fixed